### DISCUSSION TITLE
Add a 3D Gizmo NODE

### DISCUSSION THREAD

**Discussion by timpni** [godotengine/godot-proposals#9595] [3D]
Hello!
I am developing a game where users can add, remove and move 3D objects, similar to what is possible in an editor. I have created a basic 3D gizmo, but since Godot already includes a 3D Gizmo in the editor, I thought it might be more efficient to turn that into a Node.

Best,
timpni

![Gizmo into Node](https://github.com/godotengine/godot-proposals/assets/89146984/c105dca9-ae6d-45d4-935d-91e8fb8065d0)

**Comment by Calinou** [comment]
Exposing the built-in 3D gizmos to projects would mean increasing their development surface a lot, as we'd have to do things like:

- Adding many features to it to ensure it fits [everyone's use cases](https://xkcd.com/927/).
- Ensuring compatibility (and even visuals) don't change across versions, to preserve compatibility with exiting projects.

On top of that, exposing them would also increase export templates' binary size (not by a huge amount, but still).

Considering this, I don't think it's a good approach.

If something like https://github.com/godotengine/godot-proposals/issues/7213 is implemented in a manner that allows for live editing, then gizmos would need to be accessible in a running project, but they could still be compiled in debug builds only.
